Overloaded virtual function warning by clang

adam-puleo opened the pull request:
I'm compiling on a Mac and received a "warning: 'Derived::setTimeout' hides overloaded virtual function" error. From looking at [stackoverflow](https://stackoverflow.com/questions/18515183/c-overloaded-virtual-function-warning-by-clang) and the code I think this is the proper solution.

pschatzmann commented:
I could not reproduce your issue with Clang 14.0.0.
Please indicate how to reproduce this issue.

Does it make a difference if you change it to void setTimeout(unsigned long timeout)?

adam-puleo commented:
Looks like I'm running version 17:
adampuleo@Adams-M1-Pro components % clang --version
Apple clang version 17.0.0 (clang-1700.0.13.3)
Target: arm64-apple-darwin24.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

I'm on macOS 15.5 Beta.

I added arduino-audio-tools as a component in my EPS32 project. From there I just compiled the project via: idf.py build.

If I change both methods to "unsigned long" I still receive the error without the private: using BaseStream stuff that I added.

Changing both methods to "size_t" also solved the problem without the private: using BaseStream stuff that I added.

So this works:
BaseStream.h:
  /// Defines the timout the system waits for data when moving to the next stream
  void setTimeout(size_t t) { _timeout = t; }
//  private:
//   using BaseStream::setTimeout;

HttpRequest.h:
  /// Defines the client timeout in ms
  void setTimeout(size_t timeoutMs) { clientTimeout = timeoutMs; }
//  private:
//   using BaseStream::setTimeout;

pschatzmann commented:
So, in this case could you update the PR to use size_t ?
